python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -292,6 +303,14 @@ def get_latest_materialization_events(
     ) -> Mapping[AssetKey, Optional[EventLogEntry]]:
         pass
 
+    @abstractmethod
+    def get_asset_event_tags(self) -> List[Tuple[str, Set[str]]]:
```

Comment:
Do we only need this for tests? A method that returns all tags seems dangerously unperformant if a user gets a hold of it. Thoughts on scoping it to a particular asset key? Also, could use `Sequence` instead of `List` and `AbstractSet` instead of `Set`.

Reply:
That sounds good--ive changed it now